### ISSUE TITLE
Caddy v2 redirects to internal IP

### ISSUE THREAD

**Issue by samcro1967** [airsonic/airsonic#1677]
#1615  Problem description
I am not able to get reverse proxy working with Caddy V2.

When I go to my internal Caddy URL, it redirects me to the internal IP address for the host running Airsonic.

When I browse to http://local.airsonic, it redirects me to http://192.168.1.103:4040 which is the right IP and port.  I am able to login and everything works, but obviously I am not able to access it externally.

I have tried with and without "server.use-forward-headers=true" in airsonic.properties in the config folder and neither works.

## System information
Windows 10.0.18363.778
Docker  19.03.8, build afacb8b
Caddy v2.0.0-rc.3
Airsonic 10.6.1 (vias Docker image linuxserver/airsonic:latest)
Chrome Version 81.0.4044.129 (Official Build) (64-bit)

## Additional notes
**Caddyfile config:**
http://local.airsonic/ {
encode gzip
log {
	output file /logs/airsonic.log
	format single_field common_log
	}
reverse_proxy 192.168.1.103:4040 {
	header_up Host {http.reverse_proxy.upstream.hostport}
	}
}

**Comment by fxthomas** [comment]
Have you followed the [proxy docs](https://airsonic.github.io/docs/proxy/caddy/) for Caddy?

**Comment by samcro1967** [comment]
Yes, the documentation is for Caddy V1.  Caddy V2 has new capabilities and syntax.  The documentation is not valid for this use case.

**Comment by fxthomas** [comment]
Ah, thanks, I had no idea. If you feel like helping improve the docs don't hesitate :wink:

**Comment by Faeb35** [comment]
With the IIS URL Rewrite module I had to set the X-Forwarded host headers like mentioned in the nginx documentation:
https://airsonic.github.io/docs/proxy/nginx/

**Comment by samcro1967** [comment]
@Faeb35, if I understand the Caddy V2 documentation correctly, this is done automatically and by default.

"One other significant difference is that the v2 proxy passes all incoming headers thru by default (including the Host header) and sets the X-Forwarded-For header. In other words, v1's "transparent" mode is basically the default in v2 (but if you need other headers like X-Real-IP you have to set those yourself). You can still override/customize the Host header using the header_up subdirective."

**Comment by Faeb35** [comment]
Got it! 
That's probably also why, these headers were not set in the manual for Caddy V1...

**Comment by jef** [comment]
@samcro1967 is right, it's done automatically.

Here is a good example of what yours might look like:

```caddy
local.airsonic {
    encode zstd gzip
    reverse_proxy 192.168.1.103:4040
}
```

> :point_right: As a tip, using root `/` also comes for free, so you don't have to use that. The Caddy docs are pretty good, but takes a little bit to go through if you're used to v1.

**Comment by laduke** [comment]
Hey, I dunno if many people use iSub, but I think you can't use caddy 2's https with iSub
https://caddy.community/t/tls-error-from-older-ios-app/8923/3

**Comment by laduke** [comment]
Hey thanks for adding those docs! Thread got me up an running, but I have an additional thing...

I don't know if many people still use iSub, but I can't get it to work with caddy 2. iSub mostly works, but when I request an actual stream, it looks like it's using the IP Address instead of the Server name
https://caddy.community/t/tls-error-from-older-ios-app/8923/

I've tried just guessing some other header options but haven't been able to figure it out yet.
